streams/src/main/java/org/apache/kafka/streams/processor/internals/SourceNodeRecordDeserializer.java

```diff
@@ -54,4 +59,26 @@
                                     rawRecord.serializedValueSize(), key, value);
 
     }
+
+    public ConsumerRecord<Object, Object> tryDeserialize(final ProcessorContext processorContext,
+                                                         ConsumerRecord<byte[], byte[]> rawRecord) {
+
+        // catch and process if we have a deserialization handler
+        try {
+            return deserialize(rawRecord);
+        } catch (Exception e) {
+            final DeserializationExceptionHandler.DeserializationHandlerResponse response =
+                deserializationExceptionHandler.handle(processorContext, rawRecord, e);
+            if (response == DeserializationExceptionHandler.DeserializationHandlerResponse.FAIL) {
+                throw e;
```

Comment:
Should we wrap this with a `StreamsException`? This way, we could add a detailed error message and explain in the msg text that uses can set a different exception handler etc (similar to timestamp extractor?) -- so users don't need to ask at the mailing list :)